packages/timeline-state-resolver/src/conductor.ts

```diff
@@ -426,8 +440,10 @@ export class Conductor extends EventEmitter<ConductorEvents> {
 				return this.getCurrentTime()
 			}
 
+			let newDevicePromise: Promise<DeviceContainer<DeviceOptionsBase<any>> | undefined>
+
 			if (deviceOptions.type === DeviceType.ABSTRACT) {
```

Comment:
I think making use of a switch instead of this long if else chain would benefit the readability of the code. Would also remove a tiny bit of duplicate code, as `deviceOptions.type` would only need to be written ones.

The switch could/should possibly also be extracted to its own method, that simply returns the `newDevicePromise`, instead of assigning it to a variable in each case. This would again help with readability, as the method is currently 248 lines long - which is way to many.

Reply:
I extracted it and used a switch. I had some ideas on how to avoid having that switch at all (e.g. there could be some kind of a registry of devices), but it would require a lot of refactoring